packages/login_client/test/login_client_test.dart

```diff
@@ -73,6 +79,39 @@ void main() {
       },
     );
 
+    test(
+      'initialize() reads from storage, tokenEndpoint has been replaced',
+      () async {
+        final credentials = Credentials(
+          'some token',
+          tokenEndpoint: Uri.parse('https://example.com'),
+        );
+
+        when(() => credentialsStorage.read())
+            .thenAnswer((_) async => credentials);
+
+        loginClient.onCredentialsChanged.listen(
+          expectAsync1((credentials) {
+            expect(
+              credentials?.toJson(),
```

Comment:
Instead of checking `toJson` output (testing it) I'd verify the only the `credentials.tokenEndpoint`